### ISSUE TITLE
Error in top_ranking_features[[i]] : subscript out of bounds

### ISSUE THREAD

**Issue by cgpu** [cgpu/limma-nf#2]
This is fixed by 9ce5418.

Reckless before:
```r
plots_list <- list()
  for (i in 1:length(top_ranking_features)) {
      plots_list[[i]] <- drawViolin(data    = mixed_assay_pheno_data,
                                group   = "Group",
                                feature = top_ranking_features[[i]],
                                ctrl_id = "0",
                                case_id = "1")
  }
```
After sanity check:

```r
plots_list <- list()
if (length(top_ranking_features)!=0) {
  for (i in 1:length(top_ranking_features)) {
      plots_list[[i]] <- drawViolin(data    = mixed_assay_pheno_data,
                                group   = "Group",
                                feature = top_ranking_features[[i]],
                                ctrl_id = "0",
                                case_id = "1")
  }
}
```

**Comment by cgpu** [comment]
Closed by 9ce5418

**Comment by cgpu** [comment]
Closed by 9ce5418
